src/NCronJob/NCronJobExtensions.cs

```diff
@@ -33,11 +33,14 @@ public static class NCronJobExtensions
         builder.RegisterJobs(); // Complete building the NCronJobOptionBuilder
 
         services.TryAddSingleton(settings);
-        services.AddHostedService<QueueWorker>();
+        services.TryAddSingleton<QueueWorker>();
```

Comment:
Do we need `QueueWorker` somewhere else? That shouldn't be the case

Reply:
The `AddNCronJob()` method has potential to be called multiple times by an end user, so we need to make sure we're not creating _multiple_ singletons.

Reply:
`AddHostedService` is basically a `TryAddSingleton` under the hood. So I would be surprised if we really have two instances running.

Reply:
I see what you mean now. The line right after that says to wire up the QueueWorker as a HostedService. My first attempt was to get access to all of the public dictionaries so that we can surface the Job details to an API or to the tests. Since I broke up the one class into more isolated constructs we don't need this anymore. We can add a separate mechanism (separate PR) to get a snapshot of all the current Jobs, Queues and their state.